### DISCUSSION TITLE
Docs how to create custom integration

### DISCUSSION THREAD

**Discussion by michaelbe812** [reduxjs/redux-devtools#1253] [Q&A]
Hi there,

I'm wondering if there is any documentation available about how to create an extension/integration for the devtools.

With extension/integration I mean I want to provide redux-devtools for a custom library. 

Appreciate any tips =)

**Comment by markerikson** [comment]
As in, you have a state library other than Redux, and you want to know how to show that data in the Redux DevTools?

See:

- https://github.com/reduxjs/redux-devtools/blob/main/extension/docs/API/Methods.md
- https://github.com/pmndrs/zustand/blob/main/src/middleware/devtools.ts
- https://github.com/pmndrs/jotai/blob/main/src/devtools/useAtomDevtools.ts

**Reply by michaelbe812** [reply]
@markerikson those resources look great. Thanks for pointing them out

**Reply by StevenTCramer** [reply]
Are the TypeScript Types exposed for the whole API somewhere?  If so what package?

I have found interfaces (example ConnectResponse) defined multiple times and most are not even exported.  Am I missing something?

**Reply by markerikson** [reply]
They should be exported from `redux-devtools-extension`:

https://github.com/reduxjs/redux-devtools/blob/2eddfd60a8ae28f653a84e5e8e6f88c6e403636b/packages/redux-devtools-extension/src/index.ts

**Reply by Methuselah96** [reply]
(Note: the package name is now `@redux-devtools/extension`.)

**Reply by StevenTCramer** [reply]
https://github.com/reduxjs/redux-devtools/blob/2eddfd60a8ae28f653a84e5e8e6f88c6e403636b/packages/redux-devtools-extension/src/index.ts#L226

The ReduxDevtoolsExtension interface isn't exported. 
The ConnectResponse interface definition is also incomplete.

I don't understand why the `ConnectResponse` interface is defined twice in the library?

**Reply by Methuselah96** [reply]
Feel free to create a PR to export those interfaces and complete the `ConnectResponse` interface.

The types are defined in two places due to there not being a shared dependency between the `@redux-devtools/extension` package and the actual Redux DevTools Extension. There likely should be some way to share types between the two packages, but it hasn't been a priority.

**Reply by StevenTCramer** [reply]
I totally understand the priority thing, this open-source work takes a lot of time :) I'm working on my own library now.  Once I figure out what types I need I will see if I can do a PR.  I have been in the Blazor world for a while now and my TypeScript skills have diminished. :)

Thanks much for the help.
